src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -201,7 +201,11 @@ public ICompilationTracker FreezePartialStateWithTree(SolutionState solution, Do
                         var oldTree = oldState.GetSyntaxTree(cancellationToken);
 
                         compilationPair = compilationPair.ReplaceSyntaxTree(oldTree, tree);
-                        inProgressProject = inProgressProject.UpdateDocument(docState, textChanged: false, recalculateDependentVersions: false);
+
+                        // We're changing the text of this document to something else entirely.  So we need to pass
+                        // `textChanged: true, recalculateDependentVersions: true` so that any interested parties that
```

Comment:
Comment is now stale given your change to the parameters. TBH this comment also may not be necessary since this is the "obvious" logic (i.e. there really should have been a comment _before_ you'd be deleting.)

Reply:
fair.

Reply:
Removed comment.